12/umbraco-cms/reference/routing/custom-middleware.md

```diff
@@ -0,0 +1,70 @@
+---
+description: "Customizing the ASP.NET middleware pipeline in Umbraco"
+---
+
+# Middleware
+
+Middleware is a way of handling requests and responses before or after they are completed. An example can be checking if a user is logged in before adding an item to the cart. In this example it checks something before an http request is made.
```

Comment:
The middleware is responsible for actually processing/customizing the incoming request and generating the outgoing response in a pipeline (built from request delegates), so they're not something added before and after they are completed.

This is probably better explained in the first paragraph of the MS documentation: https://learn.microsoft.com/en-us/aspnet/core/fundamentals/middleware/?view=aspnetcore-7.0.